### PR TITLE
feat(channel): add presence tracking and callbacks

### PR DESCRIPTION
**Pull request by tylerbutler**
## Summary
- Add `channel.with_presence(key:, meta:)` as shorthand for tracking and sending the initial Phoenix presence state.
- Add `channel.on_presence` with an initial topic snapshot followed by ordered joins, leaves, and metadata changes. Observation works independently of tracking and includes the observing connection’s changes.
- Use the existing channel worker, atomic actor subscriptions, and bounded delivery. Stream failures close only the affected topic; rejoining obtains a fresh snapshot.
- Add lifecycle and replication coverage, guides, generated API reference, and changelog fragments.

### CODE REVIEW

File: packages/beryl/src/beryl/runtime.gleam [hunk 1]
```diff
@@ -5526,18 +5621,54 @@ fn serve(
...
+        WorkerGo(_) -> Ok(state)
+        WorkerTerminate(reason, reply) ->
+          terminate_worker(worker, link, reason, reply)
+        WorkerPresence(delivery) -> {
+          link.report(WorkerPresenceReady)
+          let next = serve_presence(worker, link, delivery)
+          case next {
+            WorkerRunning(worker, link) ->
+              Ok(WorkerInitialized(worker, link, held))
+            _ -> Ok(next)
```
Comment (tylerbutler) [inline packages/beryl/src/beryl/runtime.gleam]: **Match all variants:** This `_` catch-all over `WorkerState` disables exhaustive checking when the type changes. Match the remaining `WorkerState` variants explicitly.

File: packages/beryl/src/beryl/runtime.gleam [hunk 2]
```diff
@@ -5487,18 +5557,43 @@ fn start_worker(
...
+                    "reason",
+                    json.string(
+                      "on_presence requires a configured presence handle",
+                    ),
+                  ),
+                ]),
+              ),
+            )
+          }
+          callback, _ -> #(
```
Comment (tylerbutler) [inline packages/beryl/src/beryl/runtime.gleam]: **Match all variants:** `Option` and `Bool` are closed types, so `callback, _` bypasses exhaustive checking. Match the remaining `Some`/`None` and `True`/`False` combinations explicitly.

File: packages/beryl/src/beryl/runtime.gleam [hunk 3]
```diff
@@ -5548,21 +5679,49 @@ fn serve(
...
+        link,
+        MessageSource(
+          link.topic,
+          telemetry.PresenceMessage,
+          start_time_if(link.telemetry),
+        ),
+        Some(#(subscription, delivery.sequence)),
+        fn() { callback(delivery.event) },
+      )
+    None, _ | _, None -> WorkerRunning(worker, link)
```
Comment (tylerbutler) [inline packages/beryl/src/beryl/runtime.gleam]: **Match all variants:** Both values are closed `Option` types, so these wildcard arms lose exhaustiveness checking. Enumerate the remaining `None`/`Some` combinations explicitly.

File: packages/beryl/test/channel_presence_callbacks_test.gleam [hunk 4]
```diff
@@ -0,0 +1,522 @@
...
+    let _reply = frame(frames, "phx_reply")
+    let _diff = frame(frames, "presence_diff")
+    let _snapshot = frame(frames, "presence_state")
+    case event(trace) {
+      presence.Snapshot([]) -> {
+        let assert presence.Changed([entry], []) = event(trace)
+        entry.session_id |> should.equal("s1")
+      }
+      presence.Snapshot([entry]) -> entry.session_id |> should.equal("s1")
+      _ -> should.fail()
```
Comment (tylerbutler) [inline packages/beryl/test/channel_presence_callbacks_test.gleam]: **Match all variants:** This catch-all combines the `Changed` variant with unexpected `Snapshot` list shapes. Match `Changed(..)` and the remaining `Snapshot` shape explicitly before failing.

File: packages/beryl/src/beryl/runtime.gleam [hunk 5]
```diff
@@ -5721,14 +5889,210 @@ fn release_worker(
...
+}
+
+fn start_observing(
+  state: State(model, message),
+  socket_id: String,
+  topic: String,
+  worker: WorkerRef,
+) -> WorkerRef {
+  case worker.observer, state.config.presence {
+    None, _ -> {
```
Comment (tylerbutler) [inline packages/beryl/src/beryl/runtime.gleam]: **Match all variants:** The second value is a closed `Option`, so `None, _` bypasses exhaustive matching. Spell out `None, Some(_)` and `None, None`.

File: packages/beryl/src/beryl/runtime.gleam [hunk 6]
```diff
@@ -6684,8 +6841,13 @@ fn serve(
...
               start_time_if(link.telemetry),
             )
-          Ok(worker_step(worker, link, source, fn() { worker.on_info(mail) }))
+          Ok(
+            worker_step(worker, link, source, None, fn() {
+              worker.on_info(mail)
+            }),
+          )
         }
+        WorkerPresence(delivery) -> Ok(serve_presence(worker, link, delivery))
```
Comment (tylerbutler) [inline packages/beryl/src/beryl/runtime.gleam]: 🔴 bug: `WorkerPresence` can run while `link.current` holds another report, and `serve_presence` overwrites that reservation at L6962-L6965. The original `WorkerReportApplied` then cannot release its input reservation, so repeated presence/message overlap leaks queue capacity and eventually closes a healthy topic as overloaded. Queue presence delivery until `link.current` is `None`, then add an interleaving test that confirms queue occupancy returns to zero.
